modules/mono/glue/GodotSharp/GodotSharp/Core/Transform2D.cs

```diff
@@ -89,38 +93,31 @@ public Vector2 Scale
             }
         }
 
-        public real_t this[int rowIndex, int columnIndex]
+        /// <summary>
+        /// Access matrix elements in column-major order. The third column is the origin vector.
+        /// </summary>
+        /// <param name="column">Which column, the matrix horizontal position.</param>
+        /// <param name="row">Which row, the matrix vertical position.</param>
+        public real_t this[int column, int row]
         {
             get
             {
-                switch (rowIndex)
+                if (column == 2)
                 {
-                    case 0:
-                        return x[columnIndex];
-                    case 1:
-                        return y[columnIndex];
-                    case 2:
-                        return origin[columnIndex];
-                    default:
-                        throw new IndexOutOfRangeException();
+                    return origin[row];
                 }
+                return this[column][row];
             }
             set
             {
-                switch (rowIndex)
+                if (column == 2)
                 {
-                    case 0:
-                        x[columnIndex] = value;
-                        return;
-                    case 1:
-                        y[columnIndex] = value;
-                        return;
-                    case 2:
-                        origin[columnIndex] = value;
-                        return;
-                    default:
-                        throw new IndexOutOfRangeException();
+                    origin[row] = value;
+                    return;
                 }
+                Vector2 columnVector = this[column];
+                columnVector[row] = value;
+                this[column] = columnVector;
```

Comment:
I think the code was more readable before for this method.

Reply:
Why? Is it because the old code explicitly has the exception, while this code allows `this[column]` to throw it? I can add in an `if` for that if you wish.

Reply:
Because the index `2` is handled first. With the old code it was very clear it was x, y and origin. And since you need the if condition, code length doesn't change much.
Also this one does a copy of the vector2, but maybe the compiler optimizes that away.

Reply:
Shoot, I just realized that this isn't even necessary. Check the code I just pushed, it works.

Reply:
Looks good :)